src/main/java/com/google/sps/utilities/RealSearchDatastore.java

```diff
@@ -61,12 +61,13 @@
             PreparedQuery tutorResults = datastore.prepare(tutorQuery);
 
             for (Entity tutorEntity : tutorResults.asIterable()) {
-
+                
+                long id = (long) tutorEntity.getKey().getId();
                 String name = (String) tutorEntity.getProperty("name");
                 String email = (String) tutorEntity.getProperty("email");
                 ArrayList skills = (ArrayList) tutorEntity.getProperty("topics");
-                ArrayList<TimeRange> availability = getTimeRanges(datastore, txn, (ArrayList) tutorEntity.getProperty("availability"));
-                ArrayList<TutorSession> scheduledSessions = getScheduledSessions(datastore, txn, (ArrayList) tutorEntity.getProperty("scheduledSessions"));
+                ArrayList<TimeRange> availability = getTimeRanges(datastore, id);
```

Comment:
Shouldn't we still use getTimeRanges(datastore, txn, id) here?

Reply:
I couldn't use transactions here because they don't work with the type of query I was using. They only work with ancestor queries, so the TimeRange entities need to have the tutor entity as an ancestor. Here's more information about this: https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Ancestor_paths
I think this makes a lot of sense. I would just need to query based on the Tutor entity's key. Here's some more info about ancestor queries: https://cloud.google.com/appengine/docs/standard/java/datastore/transactions#uses_for_transactions

When we create a new TimeRange entity, we would need to specify the parent by adding the tutor entity's key as the second parameter:

`Entity time = new Entity("TimeRange", <tutor entity key>)`

I'm not sure how ancestors would work for TutorSession entities though because the sessions belong to both a tutor and a student. If I didn't use ancestors for this, I'm not sure how to use transactions for the query.

Reply:
Oh I see! That does make a lot of sense to me. I was just worried about it still being thread-safe. I'm not entirely sure whether it would be. Do you know? If not we can discuss it during pod sync today. 

As for retrieving TutorSessions, I think we could store both the tutorID and the studentID in the session. So this would mean that when we need to get a student's upcoming tutor session, we would use their id and get all the tutor sessions that have their studentID property equal to the id we passed down. Same approach for a tutor, except that in that case we would be looking at the tutorID property.

Reply:
Yeah, I'll definitely bring it up during the pod sync. 

That's what I was thinking for TutorSessions as well, but filtering by ID is not an ancestor query so I won't be able to use transactions for it. I won't be able to use transactions for any type of filtering I think (filtering by topics, email, etc.). I'm not entirely sure if we need transactions for these types of queries but we can definitely ask about it during the meeting.

Reply:
I'll go ahead an approve this PR since now we do not need to use transactions for this anymore. Thanks for looking into all of this :)